Framework/Geometry/inc/MantidGeometry/Instrument/ParameterMap.h

```diff
@@ -346,9 +347,15 @@ class MANTID_GEOMETRY_DLL ParameterMap {
 
   bool hasDetectorInfo() const;
   const Beamline::DetectorInfo &detectorInfo() const;
+  bool hasComponentInfo() const;
+  const Beamline::ComponentInfo &componentInfo() const;
   size_t detectorIndex(const detid_t detID) const;
+  const std::vector<Geometry::ComponentID> &componentIds() const;
   void
   setDetectorInfo(boost::shared_ptr<const Beamline::DetectorInfo> detectorInfo);
+  void setComponentInfo(
```

Comment:
It's a good point. As I've commented already, this is something I'd like to revisit at a later point along side some other ownership issues. I doubt this will have much impact on the performance at present, so I would suggest leaving it the way that it is.